### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by sorahn**

### CODE REVIEW

File: src/modules/BasicTH.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import PropTypes from "prop-types"
+import React from "react"
+
+/**
+ * This is the basic replacement for a TH component.
+ * If you want to use your own custom react component, instead of a raw
+ * TH component, you need to make sure it calls 'props.updateCellWidth',
+ * and sets its own width off of 'props.width'
```
Comment (lilasquared) [inline src/modules/BasicTH.js:8]: where is this TH setting its width?

Comment (sorahn) [reply src/modules/BasicTH.js:8]: the width prop goes straight through to the `th` with `...rest`

Comment (sorahn) [reply src/modules/BasicTH.js:8]: I added width as an explicit prop to better illustrate what to do.

File: src/modules/StickyTR.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import PropTypes from "prop-types"
+import React from "react"
+
+export default class StickyTR extends React.Component {
+  renderChildren = (child, columnIndex) => {
+    const width = this.props.widths[columnIndex]
+    return React.cloneElement(child, { width })
```
Comment (lilasquared) [inline src/modules/StickyTR.js:7]: so if I use my own `<th>` here and this clones it, will any props I have be lost or does `{ width }` add to my component's props?

Comment (sorahn) [reply src/modules/StickyTR.js:7]: https://reactjs.org/docs/react-api.html#cloneelement

Comment (lilasquared) [reply src/modules/StickyTR.js:7]: 👍

**Pull request merged**